### ISSUE TITLE
Inconsistent I001 behavior between macOS and Linux (same version 0.15.0)

### ISSUE THREAD

**Issue by GoldieKMG** [astral-sh/ruff#23065]
## Environment
- ruff version: 0.15.0
- macOS: passes `ruff check .`
- Linux (AWS CodeBuild): fails with I001

## ruff.toml
exclude = ["migrations", "venv", "__pycache__", ".git"]
line-length = 120

[format]
quote-style = "single"

[lint]
select = ["E", "W", "F", "I"]
ignore = []

[lint.isort]
force-single-line = true

## Code that passes on macOS but fails on Linux
import env
from common import print_message

## Linux ruff wants (blank line between import and from)
import env

from common import print_message

## Expected
Same code should pass or fail on both platforms with identical ruff version and config.

### Version

0.15.0

**Comment by lubosmj** [comment]
We experience the same issue with these wheels:
- ruff-0.15.0-py3-none-win_amd64.whl
- ruff-0.15.0-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl

This is flagged in the CI (Linux) but it is passing locally (Windows):
```
I001 [*] Import block is un-sorted or un-formatted
--> tests/test_pipeline_steps.py:1:1
|
1 | / import json
2 | | import os
3 | |
4 | | import mock
5 | | import pytest
6 | | from core import Environment, Workspace
7 | | from utilities import MockRunConfig
8 | |
9 | | from activity_training import pipeline_steps
| |_______________________________________________^
```

**Comment by ntBre** [comment]
This is often caused by a directory with the same name as a dependency being present on only one of the machines. See https://github.com/astral-sh/ruff/issues/20483 for one recent example with the `wandb` package. Is it possible that's what's happening here? I don't think we changed anything about our import detection in 0.15.0 relative to the 0.14 series.

**Comment by lubosmj** [comment]
The comment on the referenced issue says "What's happening here is that there is an XXX folder within your src folder present in one of these situations but not the other." In our case, we should have the same codebase locally and in the CI, there is no copying happening all around the place. Even with that, we would observe the problematic behaviour earlier, not after the upgrade. Did you change any rule enforcements in 0.15.0?

Our config file is pretty rudimentary:
```
exclude = ["src/Miscellaneous", "*.ipynb"]

line-length = 150

target-version = "py39"

[lint]
select = ["E4", "E7", "E9", "F"]
extend-select = [
    "BLE",  # flake8-blind-except
    "C4",  # flake8-comprehensions
    "D",  # pydocstyle
    "ICN",  # flake8-import-conventions
    "I00",  # isort
    "PL",  # pylint
    "Q",  # flake8-quotes
    "TID",  # flake8-tidy-imports
]

ignore = [
    "PLR",  # all pylint refactor rules
]

[lint.pydocstyle]
convention = "google"

[lint.extend-per-file-ignores]
"__init__.py" = [
    "D104"  # ignore undocumented-public-package
]
"!{src/ML/Libraries/utils/**.py,**/rcmlutils/**/*.py}" = [
    "D"  # ignore 'D' rules everywhere except for the utils library and rcmlutils
]
```

**Comment by lubosmj** [comment]
Hmm, thinking about it more, there is a chance that the problem is caused by us installing the "utils" and "activity_training" packages locally (see https://github.com/astral-sh/ruff/issues/10519#issuecomment-2035909837). We started doing this only recently. In the CI, it does not come pre-installed during the initial checks. So, you might be actually right!

**Comment by GoldieKMG** [comment]
re-ran the CI today without any code changes to the affected import lines, and it passed successfully.
It seems the issue may have been transient or related to caching/environment state at the time.
If this problem recurs, I will apply the suggestions from the comments above (e.g. explicitly setting known-first-party in the isort configuration).
